From 19a454a18f49252c6a8a6759f06aa69f1426405b Mon Sep 17 00:00:00 2001
From: maxwen <max.weninger@gmail.com>
Date: Sun, 25 Jun 2017 02:46:04 +0200
Subject: [PATCH] [1/3] Dialer: config to disable dnd on call

use local CALL_DND_ENABLED cause we dont want to expose
the hidden Settings value just for that

Change-Id: I0e3268f7dc8158b4cc904efffb92b4b7265dff4d
---
 res/values/custom_strings.xml                       | 21 +++++++++++++++++++++
 res/xml/sound_settings.xml                          |  6 ++++++
 .../dialer/settings/SoundSettingsFragment.java      | 14 ++++++++++++++
 3 files changed, 41 insertions(+)
 create mode 100644 res/values/custom_strings.xml

diff --git a/res/values/custom_strings.xml b/res/values/custom_strings.xml
new file mode 100644
index 000000000..95e6f902c
--- /dev/null
+++ b/res/values/custom_strings.xml
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+~ Copyright (C) 2012 The Android Open Source Project
+~
+~ Licensed under the Apache License, Version 2.0 (the "License");
+~ you may not use this file except in compliance with the License.
+~ You may obtain a copy of the License at
+~
+~      http://www.apache.org/licenses/LICENSE-2.0
+~
+~ Unless required by applicable law or agreed to in writing, software
+~ distributed under the License is distributed on an "AS IS" BASIS,
+~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+~ See the License for the specific language governing permissions and
+~ limitations under the License
+-->
+
+
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <string name="call_dnd_enabled_title">Do not disturb during call</string>
+</resources>
diff --git a/res/xml/sound_settings.xml b/res/xml/sound_settings.xml
index 879a2b0c7..70593684c 100644
--- a/res/xml/sound_settings.xml
+++ b/res/xml/sound_settings.xml
@@ -98,4 +98,10 @@
 
     </PreferenceCategory>
 
+    <CheckBoxPreference
+        android:key="call_dnd_enabled"
+        android:title="@string/call_dnd_enabled_title"
+        android:persistent="false"
+        android:defaultValue="true" />
+
 </PreferenceScreen>
diff --git a/src/com/android/dialer/settings/SoundSettingsFragment.java b/src/com/android/dialer/settings/SoundSettingsFragment.java
index 1d6ace0fd..660f6ca8a 100644
--- a/src/com/android/dialer/settings/SoundSettingsFragment.java
+++ b/src/com/android/dialer/settings/SoundSettingsFragment.java
@@ -56,10 +56,14 @@ public class SoundSettingsFragment extends PreferenceFragment
 
     private static final int MSG_UPDATE_RINGTONE_SUMMARY = 1;
 
+    private static final String KEY_CALL_DND_ENABLED = "call_dnd_enabled";
+    private static final String CALL_DND_ENABLED = "call_dnd_enabled";
+
     private Preference mRingtonePreference;
     private CheckBoxPreference mVibrateWhenRinging;
     private CheckBoxPreference mPlayDtmfTone;
     private ListPreference mDtmfToneLength;
+    private CheckBoxPreference mCallDndEnabled;
 
     private final Runnable mRingtoneLookupRunnable = new Runnable() {
         @Override
@@ -102,6 +106,12 @@ public class SoundSettingsFragment extends PreferenceFragment
         mDtmfToneLength = (ListPreference) findPreference(
                 context.getString(R.string.dtmf_tone_length_preference_key));
 
+        mCallDndEnabled = (CheckBoxPreference) findPreference(KEY_CALL_DND_ENABLED);
+        boolean dndModeEnabled = Settings.System.getInt(context.getContentResolver(),
+                CALL_DND_ENABLED, 1) == 1;
+        mCallDndEnabled.setChecked(dndModeEnabled);
+        mCallDndEnabled.setOnPreferenceChangeListener(this);
+
         if (hasVibrator()) {
             mVibrateWhenRinging.setOnPreferenceChangeListener(this);
         } else {
@@ -181,6 +191,10 @@ public class SoundSettingsFragment extends PreferenceFragment
             int index = mDtmfToneLength.findIndexOfValue((String) objValue);
             Settings.System.putInt(getActivity().getContentResolver(),
                     Settings.System.DTMF_TONE_TYPE_WHEN_DIALING, index);
+        } else if (preference == mCallDndEnabled) {
+            Boolean value = (Boolean) objValue;
+            Settings.System.putInt(getActivity().getContentResolver(),
+                    CALL_DND_ENABLED, value ? 1 : 0);
         }
         return true;
     }
-- 
2.13.2

